services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/permissions/HankeKayttajaServiceITest.kt

```diff
@@ -166,17 +161,19 @@ class HankeKayttajaServiceITest : DatabaseTest() {
     inner class GetKayttajatByHankeId {
         @Test
         fun `Returns users from correct hanke only`() {
-            val hankeToFind = hankeFactory.builder(USERNAME).withYhteystiedot().save()
-            hankeFactory.builder(USERNAME).withYhteystiedot().save()
+            val hankeToFind = hankeFactory.builder(USERNAME).save().id
+            kayttajaFactory.saveUnidentifiedUser(hankeToFind)
+            kayttajaFactory.saveUnidentifiedUser(hankeToFind, sahkoposti = "toinen@sahko.posti")
+            val otherHanke = hankeFactory.builder(USERNAME).save().id
+            kayttajaFactory.saveUnidentifiedUser(otherHanke)
+            kayttajaFactory.saveUnidentifiedUser(otherHanke, sahkoposti = "toinen@sahko.posti")
 
             val result: List<HankeKayttajaDto> =
-                hankeKayttajaService.getKayttajatByHankeId(hankeToFind.id)
+                hankeKayttajaService.getKayttajatByHankeId(hankeToFind)
 
-            assertThat(result).hasSize(5)
-            val correctIds = hankeKayttajaRepository.findByHankeId(hankeToFind.id).map { it.id }
-            assertThat(result).each { dto ->
-                dto.prop(HankeKayttajaDto::id).isIn(*correctIds.toTypedArray())
-            }
+            assertThat(result).hasSize(3)
+            val correctIds = hankeKayttajaRepository.findByHankeId(hankeToFind).map { it.id }
```

Comment:
This makes me wonder whether it would be better to have the correct ids taken from factory methods (if that is possible). If the service uses this same repository finder method.

Reply:
The problem is the founder since we can't get their ID directly. I added a database fetch to get it before creating the other users, so we know it's the founder. I think the test is better now. Thank you for the suggestion.

This also revealed a bug in `HankeBuilder.saveEntity()`. It created duplicates of the hanke.